avm/src/daemon/ctrl.rs

```diff
@@ -558,6 +567,413 @@ fn handle_keys() -> Result<Response<Body>, Infallible> {
+  let headers = req.headers();
+  let nskey = headers
+    .get("nskey")
+    .map_or("N/A", |v| v.to_str().unwrap())
+    .to_string();
+  let maybe_hm = DS.get(&nskey);
+  let subhandler_id = headers
+    .get("subhandler_id")
+    .map_or(0, |v| v.to_str().unwrap().parse().unwrap());
+  let subhandler = HandlerFragment::new(subhandler_id, 0);
+  let bytes = body::to_bytes(req.into_body()).await?;
+  let pb = protos::HandlerMemory::HandlerMemory::parse_from_bytes(&bytes)?;
+  let mut hm = HandlerMemory::from_pb(&pb)?;
+  let mut res_hm = HandlerMemory::new(None, 2)?;
+  res_hm.init_fractal(0)?;
+  match maybe_hm {
+    Some(ds) => {
+      HandlerMemory::transfer(&ds, 0, &mut hm, CLOSURE_ARG_MEM_START + 1)?;
+      let hm = subhandler.run(hm).await?;
+      res_hm.push_fixed(0, 1);
+      if hm.addr_to_idxs_opt(CLOSURE_ARG_MEM_START).is_some() {
+        // Guard against void functions
+        HandlerMemory::transfer(&hm, CLOSURE_ARG_MEM_START, &mut res_hm, 1);
+        res_hm.push_register(0, 1)?;
+      }
+      // Also grab the mutation to the datastore value and re-insert it
+      let mut newds = HandlerMemory::new(None, 1)?;
+      HandlerMemory::transfer(&hm, CLOSURE_ARG_MEM_START + 1, &mut newds, 0)?;
+      drop(ds);
+      DS.insert(nskey, newds);
+    }
+    None => {
+      res_hm.push_fixed(0, 0);
+      res_hm.push_fractal(
+        0,
+        HandlerMemory::str_to_fractal("namespace-key pair not found"),
+      )?;
+    }
+  }
+  Ok(res_hm)
+}
+
+async fn handle_dsmonly(req: Request<Body>) -> Result<Response<Body>, Infallible> {
+  match dsmonly_inner(req).await {
+    Ok(_) => Ok(Response::builder().status(200).body("ok".into()).unwrap()),
+    Err(err) => {
+      // TODO: What error message here? Also should this also be a valid HM out of here?
+      eprintln!("{:?}", err);
+      Ok(Response::builder().status(500).body("fail".into()).unwrap())
+    }
+  }
+}
+
+async fn dsmonly_inner(req: Request<Body>) -> DaemonResult<()> {
+  let headers = req.headers();
+  let nskey = headers
+    .get("nskey")
+    .map_or("N/A", |v| v.to_str().unwrap())
+    .to_string();
+  let maybe_hm = DS.get(&nskey);
+  let subhandler_id = headers
+    .get("subhandler_id")
+    .map_or(0, |v| v.to_str().unwrap().parse().unwrap());
+  let subhandler = HandlerFragment::new(subhandler_id, 0);
+  let bytes = body::to_bytes(req.into_body()).await?;
+  let pb = protos::HandlerMemory::HandlerMemory::parse_from_bytes(&bytes)?;
+  let mut hm = HandlerMemory::from_pb(&pb)?;
+  match maybe_hm {
+    Some(ds) => {
+      HandlerMemory::transfer(&ds, 0, &mut hm, CLOSURE_ARG_MEM_START + 1)?;
+      let hm = subhandler.run(hm).await?;
+      // Also grab the mutation to the datastore value and re-insert it
+      let mut newds = HandlerMemory::new(None, 1)?;
+      HandlerMemory::transfer(&hm, CLOSURE_ARG_MEM_START + 1, &mut newds, 0)?;
+      drop(ds);
+      DS.insert(nskey, newds);
+    }
+    None => {
+      // Do nothing
```

Comment:
The `only` opcodes are side-effect-only functions. They don't return anything to the consuming side (eg incrementing a counter or pushing a value onto an array).

So since the return type is `void` there's literally nothing we can do in this case where the key you're trying to mutate doesn't even exist.

It should be possible to create `refOr` and `mutOr` functions in pure alan that first check for the key existing and if it doesn't exist it sets it with the provided value if this issue is a problem for them.

In general I don't think the `mutOnly` opcodes (`dsmonly` and `dswonly`) are gonna get much use, though.

Reply:
Oh! I did not realized the returning type. Got it!